### PR TITLE
feat: add customInputStyle to TextInput

### PR DESCRIPTION
**Pull request by guilherme-ferraz1**
### Summary

Adds `customInputStyle` prop to `TextInput` as requested in #3434 
This prop allows to pass style props directly to the input itself, not the container. 

### Test plan

The tests and examples are already updated.
Tested on both IOS and Android with `StyleProp<ViewStyle>` props like `paddingLeft`, `backgroundColor` etc.

**Custom style input example:**

`customInputStyle={{ paddingLeft: 50  }}`

Captura de Tela 2022-12-06 às 23 06 18

### CODE REVIEW

File: src/components/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -143,6 +143,13 @@ export type Props = React.ComponentPropsWithRef<typeof NativeTextInput> & {
    * testID to be used on tests.
    */
   testID?: string;
+  /**
+   * @supported Available in v5.x
+   * Pass custom style directly to the input itself.
+   * Overrides input style
+   * Example: `paddingLeft`, `backgroundColor`
+   */
+  customInputStyle?: StyleProp<ViewStyle>;
```
[the remark this answers is not in the document]
Comment (gmferraz) [reply src/components/TextInput/TextInput.tsx]: sure

File: src/components/TextInput/TextInput.tsx [hunk 2]
```diff
@@ -226,6 +233,7 @@ const TextInput = React.forwardRef<TextInputHandles, Props>(
       error: errorProp = false,
       multiline = false,
       editable = true,
+      customInputStyle = null,
```
[the remark this answers is not in the document]
Comment (gmferraz) [reply src/components/TextInput/TextInput.tsx]: fixed

**Comment by gmferraz** [comment]
@lukewalczak unit test added, prop name changed and minor issues fixed

**Comment by lukewalczak** [comment]
Hey @gmferraz, I've analyzed your PR one more time and have two requests, please:

* resolve conflict
* rename prop to `contentStyle` - it's more appropriate than `containerStyle` proposed previously.

I would like to do the release tomorrow, so please handle it if possible 🙏🏽

**Comment by gmferraz** [comment]
@lukewalczak done :)

File: src/components/TextInput/TextInputFlat.tsx [hunk 3]
```diff
@@ -149,7 +150,7 @@ const TextInputFlat = ({
     theme,
   });
 
-  const containerStyle = {
+  const contentStyle = {
```
Comment (lukewalczak) [inline src/components/TextInput/TextInputFlat.tsx:153]: I think it should not be changed.

File: src/components/TextInput/TextInputFlat.tsx [hunk 4]
```diff
@@ -307,7 +309,7 @@ const TextInputFlat = ({
   }
 
   return (
-    <View style={[containerStyle, viewStyle]}>
+    <View style={[contentStyle, viewStyle]}>
```
Comment (lukewalczak) [inline src/components/TextInput/TextInputFlat.tsx:312]: The same here.

File: src/components/TextInput/TextInputFlat.tsx [hunk 5]
```diff
@@ -336,7 +338,7 @@ const TextInputFlat = ({
               dense ? styles.densePatchContainer : styles.patchContainer,
               {
                 backgroundColor:
-                  viewStyle.backgroundColor || containerStyle.backgroundColor,
+                  viewStyle.backgroundColor || contentStyle.backgroundColor,
```
Comment (lukewalczak) [inline src/components/TextInput/TextInputFlat.tsx:341]: And there as well.

File: src/components/TextInput/TextInputFlat.tsx [hunk 6]
```diff
@@ -73,6 +73,7 @@ const TextInputFlat = ({
   right,
   placeholderTextColor,
   testID = 'text-input-flat',
+  contentStyle: customContainerStyle,
```
Comment (lukewalczak) [inline src/components/TextInput/TextInputFlat.tsx:76]:
```suggestion
  contentStyle,
```

File: src/components/TextInput/TextInputFlat.tsx [hunk 7]
```diff
@@ -380,6 +382,7 @@ const TextInputFlat = ({
             },
             Platform.OS === 'web' && { outline: 'none' },
             adornmentStyleAdjustmentForNativeInput,
+            customContainerStyle,
```
Comment (lukewalczak) [inline src/components/TextInput/TextInputFlat.tsx:385]:
```suggestion
            contentStyle,
```

**Review by lukewalczak — CHANGES_REQUESTED** [review]

File: src/components/TextInput/TextInputFlat.tsx (hunk 7 above)
Comment (gmferraz) [reply src/components/TextInput/TextInputFlat.tsx:385]: oops I guess it should be fine now

**Review by lukewalczak — APPROVED** [review]

**Pull request merged**